src/cdk/tree/tree.ts

```diff
@@ -146,6 +149,16 @@ export class CdkTree<T, K = T> implements AfterContentChecked, CollectionViewer,
    */
   @Input() expansionKey?: (dataNode: T) => K;
 
+  /**
+   * What type of node is being used in the tree. This must be provided if either of
+   * `levelAccessor` or `childrenAccessor` are provided.
+   *
+   * This controls what selection of data the tree will render.
+   */
+  // NB: we're unable to determine this ourselves; Angular's ContentChildren
+  // unfortunately does not pick up the necessary information.
+  @Input() nodeType?: 'flat' | 'nested';
```

Comment:
Why is this necessary at the node level? I would think that this is only needed at the entire tree level